packages/common/src/Logger.ts

```diff
@@ -23,24 +23,33 @@ enum LOGGING_LEVELS {
 }
 
 export default class Logger implements ILogger {
-  private logger: WinstonLogger
+  private _logger: WinstonLogger
+  private logLabel: string
 
   private readonly format = printf(({ level, message, label, timestamp }) => {
     return `${timestamp} [${label}] ${level}: ${message} `
   })
 
   constructor(logLabel: string) {
-    try {
-      switch (configLoader().LOGGING_MODE) {
-        case 'GCP':
-          this.logger = this.getGCPLogger()
-          break
-        default:
-          this.logger = this.getDefaultLogger(logLabel)
+    this.logLabel = logLabel
+  }
+
+  get logger() {
```

Comment:
I'm not sure we understand why this refactoring is required to allow you to use a different logger in the backstage plugin. It looks like it's just a refactor for now... Are these changes going to make using the backstage logger easier in some way?

Reply:
There were a few places around the code where the config was being used at the point the code was being executed. We've had to make these changes to delay the request for the config so that there is time during the setup of the router to set the config.